### DISCUSSION TITLE
Can filter a records based on inner multiple relational list of items?

### DISCUSSION THREAD

**Discussion by devmuaz** [pocketbase/pocketbase#3490] [Q&A] [answered]
Hello, recently I made some changes to a collection I have, which contains a multiple relational field, my question is that can I filter these records based on those inner fields?

The current response:

```json
{
    "page": 1,
    "perPage": 30,
    "totalPages": 1,
    "totalItems": 2,
    "items": [
        {
            "id": "RECORD_ID",
            "collectionId": "y7ovpkz74gyh2v5",
            "collectionName": "apps",
            "created": "2022-01-01 01:00:00.123Z",
            "updated": "2022-01-01 23:59:59.456Z",
            "expand": {
                "localization": [
                    {
                        "id": "RECORD_ID",
                        "collectionId": "j1hepslwkzdnzqr",
                        "collectionName": "appsLocalization",
                        "created": "2022-01-01 01:00:00.123Z",
                        "updated": "2022-01-01 23:59:59.456Z",
                        "name": "test",
                        "title": "test",
                        "locale": "test"
                    },
                    {
                        "id": "RECORD_ID",
                        "collectionId": "j1hepslwkzdnzqr",
                        "collectionName": "appsLocalization",
                        "created": "2022-01-01 01:00:00.123Z",
                        "updated": "2022-01-01 23:59:59.456Z",
                        "name": "test",
                        "title": "test",
                        "locale": "test"
                    }
                ]
            },
            "name": "test",
            "image": "filename.jpg",
            "localization": [
                "RELATION_RECORD_ID"
            ],
            "category": "RELATION_RECORD_ID",
            "appInfoId": "RELATION_RECORD_ID",
            "ios": true,
            "android": true
        }
    ]
}
```

I tried to write a filter condition: `localization.locale = "en"`, but got no records although there're indeed items have localization items with `en locale`.

Call: `/api/collections/items/records?filter=localization.locale="en"`

Is this possible to do?

This is helpful when doing a search query in a search feature!

**Comment by ganigeorgiev** [comment, marked as answer]
You can use dot-notation to target nested relation fields, but keep in mind that the filtering will apply only for the main collection records. You can't filter the `expand` results if that is what you are trying to do.

Your example should work if you prefix the equal sign with `?`.

```js
await pb.collection("items").getFullList({
    filter: 'localization.locale ?= "en"'
})
```

The above returns all `items` records that have any/at-least-one `localization` related record with locale equal to "en".

**Reply by devmuaz** [reply]
Yes, exactly that worked for me!
I also tried the `?~` - `Like/Contains` for other fields like `title` or `name` and worked too!

Thanks mate, much appreciated ❤️
